src/basic/in-addr-util.c

```diff
@@ -226,6 +226,70 @@ int in_addr_prefix_next(int family, union in_addr_union *u, unsigned prefixlen)
         return -EAFNOSUPPORT;
 }
 
+int in_addr_prefix_nth(int family, union in_addr_union *u, unsigned prefixlen, unsigned nth) {
+        assert(u);
+
+        if (prefixlen <= 0)
+                return 0;
+
+        if (nth == 0) { // identity??
+                return 1;
+        }
```

Comment:
coding style: we generally don't place {} around single-line if blocks, see formatting